pkg/aws/cloud.go

```diff
@@ -44,9 +47,28 @@ type Cloud interface {
 
 // NewCloud constructs new Cloud implementation.
 func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud, error) {
+	hasIPv4 := true
+	addrs, err := net.InterfaceAddrs()
+	if err == nil {
```

Comment:
in case of error, why not return right away?
Are there situations where the net.InterfaceAddrs() return error, but we still want to continue processing?

Reply:
The idea was to avoid any negative effects in existing cases. So if the process doesn't have permission to enumerate the interfaces, it would assume IPv4.